[RNMobile] Add a subtitle for unsupported blocks

maxme opened the pull request:
## Description

Add a subtitle for unsupported blocks that we know the names of.

## How has this been tested?
<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, tests ran to see how -->
<!-- your change affects other areas of the code, etc. -->

## Screenshots <!-- if applicable -->

Light mode (Android)
<img width="410" src="https://user-images.githubusercontent.com/40213/67564809-1e417380-f724-11e9-80bc-bcbb5bd6879b.png">

Dark mode (iOS)
<img width="410" alt="Screenshot 2019-10-25 at 15 18 35" src="https://user-images.githubusercontent.com/40213/67574546-eeea3100-f73a-11e9-99a7-028dffe0c7d3.png">



cc @iamthomasbishop - can you double check the font size and font color match your design document?

## Types of changes

This is a new feature (non-breaking change which adds functionality). Only affects mobile apps.

## Checklist:
- [x] My code is tested (manual tests only).
- [x] Styling supports dark mode on iOS.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->

maxme committed:
Add a new unsupported subtitle to missing blocks - even we know about the block title

iamthomasbishop commented:
Had a chance to double-check the design here. Looks pretty good, just a few requests:

* `margin-top` on `.unsupportedBlockMessage` == `4`
* `margin-top` on `.unsupportedBlockSubtitle` == `2`
* Colors are fine in light mode
* In dark mode, `. unsupportedBlockSubtitle` can == `gray 20` from [Color Studio](https://color-studio.blog) (title can be white same as your screenshots)


<img width="411" alt="image" src="https://user-images.githubusercontent.com/1200432/67586373-1929f780-f717-11e9-88d4-b695e1b861e0.png">

iamthomasbishop commented:
One more thing I forgot to mention, can we also make `.unsupportedBlockMessage` use the Medium font-weight (which I think is 600)?

maxme committed:
Update margins, colors and font weight of the unsupported block

maxme commented:
Here is the updated screenshot with the changes of margins, subtitle color and font-weigth:

![600](https://user-images.githubusercontent.com/40213/67617269-b94c5300-f7e1-11e9-8d37-aab7a59170ee.png)

>     margin-top on .unsupportedBlockMessage == 4

Updated in d95fe07a8

>     margin-top on .unsupportedBlockSubtitle == 2

Updated in d95fe07a8

>    In dark mode, . unsupportedBlockSubtitle can == gray 20 from Color Studio (title can be white same as your screenshots)

Updated in d95fe07a8 - using gray-20: `#a7aaad`

> One more thing I forgot to mention, can we also make `.unsupportedBlockMessage` use the Medium font-weight (which I think is 600)?

Updated in d95fe07a8 using `font-weight: 600`.

I had a look at other blocks:
- On the media placeholder, the font-weight for empty view description is 500.
- On the bottom sheet, the font-weight title is 600.

I'm not sure if we have a well defined "medium" value, so I took some screenshots with default, 500 and 600:

Default | font-weight: 500 | font-weight: 600
-|-|-
![1-default-cropped](https://user-images.githubusercontent.com/40213/67617168-b735c480-f7e0-11e9-91d6-eabb60a9d12f.png) |  ![500-cropped](https://user-images.githubusercontent.com/40213/67617169-b7ce5b00-f7e0-11e9-8b71-ebc6412f8b6b.png) |  ![600-cropped](https://user-images.githubusercontent.com/40213/67617170-b7ce5b00-f7e0-11e9-943a-049a5b1d8b8c.png)

(Full sized screenshots: https://bia.is/s/yCa5/)

iamthomasbishop commented:
Looks great! :shipit: 

> I had a look at other blocks:
> 
> On the media placeholder, the font-weight for empty view description is 500.
> On the bottom sheet, the font-weight title is 600.
>
> I'm not sure if we have a well defined "medium" value, so I took some screenshots with default, 500 and 600:

That makes sense – Roboto and SF Pro also have _slightly_ different visual weighting, but let's roll with 600.

SergioEstevao reviewed (approved):
Code looks good, and it's working fine! :shipit:

hypest commented on packages/block-library/src/missing/edit.native.js:
```diff
@@ -27,13 +27,17 @@ export class UnsupportedBlockEdit extends Component {
 		const titleStyle = getStylesFromColorScheme( styles.unsupportedBlockMessage, styles.unsupportedBlockMessageDark );
 
+		const subTitleStyle = getStylesFromColorScheme( styles.unsupportedBlockSubtitle, styles.unsupportedBlockSubtitleDark );
+		const subtitle = blockType ? <Text style={ subTitleStyle }>{ __( 'Unsupported' ) }</Text> : null;
```
👋 @maxme , not sure I'm missing context, why don't we display the `Unsupported` subtitle for all unsupported blocks?

maxme replied:
The title for unsupported block we don't know the name is currently `Unsupported`. If we showed the subtitle, we would show `Unsupported` twice.

See the last unsupported block on this screenshot:

<img width="410" alt="Screenshot 2019-10-25 at 15 18 35" src="https://user-images.githubusercontent.com/40213/67574546-eeea3100-f73a-11e9-99a7-028dffe0c7d3.png">
